services/export/shared/transactionsExport.js

```diff
@@ -0,0 +1,555 @@
+				const timestampRange = `${fromTimestamp}:${toTimestamp}`;
+				const transactions = await requestAllStandard(
+					getTransactionsInAsc,
+					{ ...params, timestamp: timestampRange },
+					MAX_NUM_TRANSACTIONS,
+				);
+				allTransactions.push(...transactions);
+
+				const incomingCrossChainTransferTxs = await getCrossChainTransferTransactionInfo({
+					...params,
+					timestamp: timestampRange,
+				});
+
+				const rewardAssignedInfo = await getRewardAssignedInfo({
+					...params,
+					timestamp: timestampRange,
+				});
+
+				if (incomingCrossChainTransferTxs.length || rewardAssignedInfo.length) {
+					allTransactions.push(...incomingCrossChainTransferTxs, ...rewardAssignedInfo);
+					allTransactions.sort((a, b) => a.block.height - b.block.height);
+				}
+
+				if (day !== getToday()) {
+					if (transactions.length) {
+						partials.write(partialFilename, JSON.stringify(allTransactions));
+					} else {
+						// Flag to prevent unnecessary calls to core/storage space usage on the file cache
+						const RETENTION_PERIOD_MS = getDaysInMilliseconds(
+							config.cache.partials.retentionInDays,
+						);
+						await noTransactionsCache.set(partialFilename, true, RETENTION_PERIOD_MS);
+					}
+				}
+			}
+			/* eslint-enable no-await-in-loop */
+		}
+	}
+
+	const excelFilename = await getExcelFilenameFromParams(params);
+
+	// Add duplicate entry with zero fees for self token transfer transactions
+	allTransactions.forEach((tx, i, arr) => {
+		if (checkIfSelfTokenTransfer(tx) && !tx.isSelfTokenTransferCredit) {
+			arr.splice(i + 1, 0, { ...tx, fee: '0', isSelfTokenTransferCredit: true });
+		}
+	});
+
+	const currentChainID = await getCurrentChainID();
+	const normalizedTransactions = await Promise.all(allTransactions
+		.map((t) => normalizeTransaction(getAddressFromParams(params), t, currentChainID)));
+	const metadata = await getMetadata(params, currentChainID);
+
+	const workBook = new excelJS.Workbook();
+	const transactionExportSheet = workBook.addWorksheet(config.excel.sheets.TRANSACTION_HISTORY);
+	const metadataSheet = workBook.addWorksheet(config.excel.sheets.METADATA);
+	transactionExportSheet.columns = fields.transactionMappings;
+	metadataSheet.columns = fields.metadataMappings;
+
+	transactionExportSheet.addRows(normalizedTransactions);
+	metadataSheet.addRows([metadata]);
+
+	await workBook.xlsx.writeFile(`${config.cache.exports.dirPath}/${excelFilename}`);
+};
+
+const scheduleTransactionExportQueue = Queue(
+	config.endpoints.redis,
+	config.queue.scheduleTransactionExport.name,
+	exportTransactions,
+	config.queue.scheduleTransactionExport.concurrency,
+);
+
+const scheduleTransactionHistoryExport = async (params) => {
+	// Schedule only when index is completely built
+	const isBlockchainIndexReady = await requestIndexer('isBlockchainFullyIndexed');
+	if (!isBlockchainIndexReady) throw new ValidationException('The blockchain index is not yet ready. Please retry later.');
```

Comment:
This error might be very strange to the end user, since the indexing may not always be 100% due to various hardware or latency issues.
Should we maybe proceed if indexing is somewhere close to 100%?

Reply:
No. If we do not have the indexing at 100%, then we'll end up serving incorrect information that the financial auditors would then rely on.